lib/puppet/type/rhsm_reconfigure_script.rb

```diff
@@ -0,0 +1,31 @@
+Puppet::Type.newtype(:rhsm_reconfigure_script) do
+  desc 'Creates script to reconfigure RHSM to point at a server'
+
+  ensurable
+
+  newparam(:name, :namevar => true)
```

Comment:
The name is really the path, right? You don't need the namevar to be `name`: https://puppet.com/docs/puppet/5.5/custom_types.html#namevar